### ISSUE TITLE
Reference to system property not replaced in @ActivationConfigProperty annotation

### ISSUE THREAD

**Issue by c0ffe3** [payara/Payara#2075]
# Description #
------------

Inside the @MessageDriven annotation of a MDB, I want to specify the name of the resource adapter through a system variable (as described in the [documentation](https://docs.payara.fish/documentation/core-documentation/configuration/var-substitution/types-of-variables.html)). The relevant part is the following:

```
@MessageDriven(activationConfig = {
	@ActivationConfigProperty(propertyName = "resourceAdapter", propertyValue ="${my.system.variable}")
})
```
When I replace the variable through the String representation of my local resource adapter name, the annotation works without problems.

The system variable is set by executing the following command while the system is running, and restarting afterwards:
`asadmin create-system-properties my.system.variable=value`

## Current Outcome

I always get the same error: 
```
[2017-10-17T14:27:08.152+0200] [Payara 4.1] [SEVERE] [] [javax.enterprise.system.core] [tid: _ThreadID=49 _ThreadName=admin-thread-pool::admin-listener(1)] [timeMillis: 1508243228152] [levelValue: 1000] [[
  Exception while loading the app : EJB Container initialization error
com.sun.appserv.connectors.internal.api.ConnectorRuntimeException: Resource adapter ${my.system.variable} is not deployed
	at com.sun.enterprise.connectors.inbound.ConnectorMessageBeanClient.getActiveResourceAdapter(ConnectorMessageBeanClient.java:321)
	at com.sun.enterprise.connectors.inbound.ConnectorMessageBeanClient.setup(ConnectorMessageBeanClient.java:166)
	at org.glassfish.ejb.mdb.MessageBeanContainer.<init>(MessageBeanContainer.java:236)
	at org.glassfish.ejb.mdb.MessageBeanContainerFactory.createContainer(MessageBeanContainerFactory.java:63)
	at org.glassfish.ejb.startup.EjbApplication.loadContainers(EjbApplication.java:225)
        ...
]]

```

## Context

I ran some tests and found out that the system variable can be obtained inside my application code by executing System.getProperty("my.system.variable"). For this to work, I could set the variable during domain setup by any of the following commands:

```
asadmin create-jvm-options -Dmy.system.variable=value
asadmin create-system-properties my.system.variable=value
asadmin create-system-properties --target domain my.system.variable=value
```

When testing different calls to System.getenv(), I never received anything besides null. Tested input formats included:
("my.system.variable"),
("${my.system.variable}"),
("-Dmy.system.variable"),
("ENV=my.system.variable"),
("ENV=\\my.system.variable").

## Environment ##

- **Payara Version**: 4.1.2.173
- **Edition**: Full
- **JDK Version**: 1.8.0_122 - Oracle
- **Operating System**: Windows

**Comment by Cousjava** [comment]
To refer to an environment variable use `propertyValue ="${ENV=my.system.variable}")`

**Comment by c0ffe3** [comment]
Using `propertyValue ="${ENV=my.system.variable}")` results in the same error as previously.
Also, the variable is logged out at server start, so I'd assume it is set correctly and the issue occurs during the replacement:
```
[INFO] [NCLS-GFLAUNCHER-00005] [javax.enterprise.launcher] [tid: _ThreadID=1 _ThreadName=main] [timeMillis: 1508257538690] [levelValue: 800] [[
  JVM invocation command line:
  ...
  -Dmy.system.variable=value
  ...
]]
```

**Comment by Cousjava** [comment]
I have managed to reproduce your issue, it only occurs for the resourceAdapter property. The way you try to access it in your orginal code is the correct way to do so for system variables.

Internal issue number PAYARA-2221 created for it.

**Comment by smillidge** [comment]
Fixed by #2189
